src/main/java/com/marklogic/contentpump/SplitDelimitedTextReader.java

```diff
@@ -250,4 +252,79 @@ protected String convertToLine(String[] values) {
         return sb.toString();
     }
 
+    /*
+      Bug 55944: Adjust the starting position of a split based on encoding, in
+      case the split boundary is in the middle of a multibyte character.
+     */
+    private long adjustStartPos()
+            throws IllegalArgumentException, IOException {
+        if (start == 0) return start;
+        // Only support UTF-8 for now
+        if (encoding.equals("UTF-8")) return findStartPosUTF8();
+        else return start;
+    }
+
+    private long findStartPosUTF8()
+            throws IllegalArgumentException, IOException {
+        long curPos = start;
+        int numBytes = 4; // UTF-8 encoded characters have maximum 4 bytes
+        while (fileIn.available() > 0 && curPos > 0) {
+            // Unable to find byte 1 of a UTF-8 char within 4 moves
+            if ((start-curPos) >= numBytes)
+                throw new IllegalArgumentException("Invalid UTF-8 encoding");
+            fileIn.seek(curPos-1);
+            int prevByte = fileIn.read();
+            int curByte = fileIn.read();
+            if (checkFirstByteUTF8(curByte)) { // Byte 1 found
+                // Check for special case: the split boundary is right after
+                // the newline char or right before a delimiter
+                if (prevByte == (int)('\n') || prevByte == (int)('\r'))
+                    // Split boundary right after the newline char, backoff by
+                    // 1 byte to create a partial line
+                    return curPos-1;
+                else return curPos + delimiterOffsetUTF8(delimiter, curPos);
+            }
+            else curPos--;
+        }
+        return start;
+    }
+
+    private boolean checkFirstByteUTF8(int curByte)
```

Comment:
You could probably do some bitwise operation by Anding currentbyte with 1100000, if the result = 1000000 =128 then it's not the start of a utf8 char. The current logic is fine though. thanks